### DISCUSSION TITLE
A new name for storage engine?

### DISCUSSION THREAD

**Discussion by skyzh** [risinglightdb/risinglight#470] [Ideas]
The current on-disk storage engine is called `secondary`. That's a long story to be told...

As @MingjiHan99 bootstrapped the RisingLight project, we expect to implement a SQLite-like on-disk storage engine -- single file, update in-place, textbook concurrency control. But later I thought the old-school storage engine wasn't a good fit for a modern educational database. At that time, I came up with a new design, the merge-tree column store engine. Everything on disk is immutable, which greatly simplified the concurrency control in our system.

Months ago, there were two folders in `storage`: `disk`, `memory`. How should we call the second storage engine? It turned out to be `secondary`, and we planned to develop all of the three engines, and make a shared trait for them to support adaptive storage engine for RisingLight.

Later, we decided not to continue with the `disk` engine development. So now we only have two engines -- `memory` and `secondary`.

And I believe now it's a good time to determine a new name for `secondary`. If you have any idea, feel free to post in this discussion.

**Comment by xxchan** [comment]
I finally know why it is called `secondary` 🤣🤣🤣
